homeassistant/components/motioneye/config_flow.py

```diff
@@ -43,79 +42,94 @@ async def async_step_user(
-                    )
-                    return out
-
-        out = self.async_show_form(
-            step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_URL, default=user_input.get(CONF_URL, "")): str,
-                    vol.Optional(
-                        CONF_ADMIN_USERNAME, default=user_input.get(CONF_ADMIN_USERNAME)
-                    ): str,
-                    vol.Optional(
-                        CONF_ADMIN_PASSWORD, default=user_input.get(CONF_ADMIN_PASSWORD)
-                    ): str,
-                    vol.Optional(
-                        CONF_SURVEILLANCE_USERNAME,
-                        default=user_input.get(CONF_SURVEILLANCE_USERNAME),
-                    ): str,
-                    vol.Optional(
-                        CONF_SURVEILLANCE_PASSWORD,
-                        default=user_input.get(CONF_SURVEILLANCE_PASSWORD),
-                    ): str,
-                }
-            ),
-            errors=errors,
+            return _get_form(reauth_entry.data if reauth_entry else {})
+
+        try:
+            # Cannot use cv.url validation in the schema itself, so
+            # apply extra validation here.
+            cv.url(user_input[CONF_URL])
+        except vol.Invalid:
+            return _get_form(user_input, {"base": "invalid_url"})
+
+        client = create_motioneye_client(
+            user_input[CONF_URL],
+            admin_username=user_input.get(CONF_ADMIN_USERNAME),
+            admin_password=user_input.get(CONF_ADMIN_PASSWORD),
+            surveillance_username=user_input.get(CONF_SURVEILLANCE_USERNAME),
+            surveillance_password=user_input.get(CONF_SURVEILLANCE_PASSWORD),
+        )
+
+        try:
+            await client.async_client_login()
+        except MotionEyeClientConnectionError:
+            return _get_form(user_input, {"base": "cannot_connect"})
+        except MotionEyeClientInvalidAuthError:
+            return _get_form(user_input, {"base": "invalid_auth"})
+        except MotionEyeClientRequestError:
+            return _get_form(user_input, {"base": "unknown"})
+        finally:
+            await client.async_client_close()
+
+        if self.context.get(CONF_SOURCE) == SOURCE_REAUTH and reauth_entry is not None:
+            self.hass.config_entries.async_update_entry(reauth_entry, data=user_input)
+            # Need to manually reload, as the listener won't have been
+            # installed because the initial load did not succeed (the reauth
+            # flow will not be initiated if the load succeeds).
+            await self.hass.config_entries.async_reload(reauth_entry.entry_id)
+            out = self.async_abort(reason="reauth_successful")
+            return out
+
+        # Search for duplicates: there isn't a useful unique_id, but
+        # at least prevent entries with the same motionEye URL.
+        for existing_entry in self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
We can use `self._async_current_entries`

https://github.com/home-assistant/core/blob/49c23bad29785fd541f9c2e0c09ae3dad795f22a/homeassistant/components/enphase_envoy/config_flow.py#L94-L101